### PR TITLE
Event at end of day bugfix + auto textcolor for events

### LINKED ISSUE #29: All Events are cleared when creating new event at end of day
In Sample app:

1. Click somewhere near the end of the day
2. All events are cleared

### PR DESCRIPTION
**Pull request by wietsebuseyne**
Fixes #29 
Adds possibility to select textcolor based on Event
Changes chosen start time for new events: new event added around tapped position instead of after

### CODE REVIEW

File: library/src/main/java/com/alamkanak/weekview/WeekViewUtil.java [hunk 1]
```diff
@@ -67,4 +69,10 @@ public static int daysBetween(Calendar dateOne, Calendar dateTwo){
                 (dateOne.getTimeInMillis() + dateOne.getTimeZone().getOffset(dateOne.getTimeInMillis())))
                         / (1000 * 60 * 60 * 24f));
     }
+
+    public static int getTextColor(int color) {
```
Comment (jhoobergs) [inline library/src/main/java/com/alamkanak/weekview/WeekViewUtil.java]: I think this function shouldn't be included in the weekview. I think it should be possible to implement this method with an interface, so everyone can use it's own implementation.

Comment (jhoobergs) [reply library/src/main/java/com/alamkanak/weekview/WeekViewUtil.java]: @wietsebuseyne

Comment (wietsebuseyne) [reply library/src/main/java/com/alamkanak/weekview/WeekViewUtil.java]: You are right, I will change this

**Comment by wietsebuseyne** [comment]
Added TextColorPicker interface to choose textcolor based on the event

**Review by jhoobergs — COMMENTED** [review]
@wietsebuseyne This seems ok, I'll test it during next week and then make a new release.

File: library/src/main/java/com/alamkanak/weekview/WeekView.java [hunk 2]
```diff
@@ -953,6 +962,8 @@ private Calendar getTimeFromPoint(float x, float y){
                         - mHeaderRowPadding * 2 - mTimeTextHeight/2 - mHeaderMarginBottom;
                 int hour = (int)(pixelsFromZero / mHourHeight);
                 int minute = (int) (60 * (pixelsFromZero - hour * mHourHeight) / mHourHeight);
+                minute -= Math.min(minute,  (mNewEventLengthInMinutes / 2));
```
Comment (jhoobergs) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: I think this shouldn't always happen, this method is also used at other places where the length of new events doesn't make any difference.

**Review by jhoobergs — CHANGES_REQUESTED** [review]
After changing my small comment, can you rebase your branch to the develop branch of this repo ?

File: library/src/main/java/com/alamkanak/weekview/WeekView.java [hunk 3]
```diff
@@ -1085,6 +1096,9 @@ private void drawEventTitle(WeekViewEvent event, RectF rect, Canvas canvas, floa
         int availableWidth = (int) (rect.right - originalLeft - mEventPadding * 2);
 
         // Get text dimensions.
+        if(textColorPicker != null) {
```
Comment (jhoobergs) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: This if should be placed at line 554 or something, now it's set every time a title is drawn

Comment (wietsebuseyne) [reply library/src/main/java/com/alamkanak/weekview/WeekView.java]: This is the desired behavior I think, as each event can require a different textColor (e.g. events with a dark background gets white text and events with a white background get black text)

If it is set in the init() function, all events will have the same text color.

Comment (jhoobergs) [reply library/src/main/java/com/alamkanak/weekview/WeekView.java]: Yes, that's right. I was thinking about how to use it with the setTextColor methods, but that isn't possible. So this is right.

**Comment by wietsebuseyne** [comment]
Rebased to merge the changes with the start- and endTime

**Comment by jhoobergs** [comment]
See #78 (I rebased there and I'm fixing some other things)

**Pull request closed without merging**